services/QuillLMS/spec/middleware/canvas_auth_spec.rb

```diff
@@ -0,0 +1,40 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Auth::Canvas::Setup do
+  subject { described_class.new(env, strategy: strategy).run }
+
+  let(:canvas_instance) { create(:canvas_config).canvas_instance }
+  let(:rack_app) { ->(env) { [200, {}, ''] } }
+  let(:strategy) { OmniAuth::Strategies::Canvas.new(rack_app, {}) }
+
+  describe 'request phase' do
```

Comment:
Is it worth also testing setup and teardown logic here?  Or somewhere?

Reply:
This is a good call out.  I've been debating on this piece for a little bit.  It's kind of messy/clunky to test in the unit spec itself.  I'm probably going to add something in an system test in a subsequent PR.